### PR TITLE
refactor: add test to verify update event handle

### PR DESCRIPTION
**Pull request by amarlearning**
Add test to verify update event handled by CRI-O

**Does this PR introduce a user-facing change?**
```
None
```

### CODE REVIEW

File: pkg/ocicni/ocicni.go [hunk 1]
```diff
@@ -155,9 +155,9 @@ func (plugin *cniNetworkPlugin) monitorConfDir(start *sync.WaitGroup) {
 			logrus.Infof("CNI monitoring event %v", event)
 
 			var defaultDeleted bool
-			createWriteRename := (event.Op&fsnotify.Create == fsnotify.Create ||
+			createWriteRename := event.Op&fsnotify.Create == fsnotify.Create ||
 				event.Op&fsnotify.Write == fsnotify.Write ||
-				event.Op&fsnotify.Rename > 0)
+				event.Op&fsnotify.Rename > 0
```
Comment (haircommander) [inline pkg/ocicni/ocicni.go:160]: on reflection, shouldn't this be
`event.Op&fsnotify.Rename == fsnotify.Rename`
to match the others

File: pkg/ocicni/ocicni_test.go [hunk 2]
```diff
@@ -5,21 +5,19 @@ import (
...
 	"github.com/containernetworking/cni/pkg/version"
 	"github.com/containernetworking/plugins/pkg/ns"
 	"github.com/containernetworking/plugins/pkg/testutils"
+	"github.com/vishvananda/netlink"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
```
Comment (haircommander) [inline pkg/ocicni/ocicni_test.go:20]: these should go above so the stdlibrary imports are grouped

File: pkg/ocicni/ocicni.go (hunk 1 above)
Comment (amarlearning) [reply pkg/ocicni/ocicni.go]: yes 👍🏻

**Comment by amarlearning** [comment]
@haircommander the network config gets updated automatically when we updated the config file

**Comment by amarlearning** [comment]
@haircommander I have added the test to verify that when network config file changes, the CRI-O network plugin automatically picks up new config.

**Comment by haircommander** [comment]
/approve

 so has CRI-O always caught `modify` events or is it just now doing so? After we merge this, we can maybe add a bats test in CRI-O to check there as well e2e

@cri-o/ocicni-maintainers PTAL

**Comment by amarlearning** [comment]
@haircommander I looked at the git history to cross-check on this and it seems like the `modify` event code has been there for some time.

I think it makes sense to now add a bats test in CRI-O to check there as well e2e.

**Comment by saschagrunert** [comment]
@amarlearning are you going to take care of this? :)

**Comment by amarlearning** [comment]
inotify wirte events do tell us if a new file has been created or if there is an update in the existing file. Not sure if this was there before, they might have recently added the update events in the write method which eventually resolved the issue.

**Comment by amarlearning** [comment]
I went through the release notes of both `inotify` and `fsnotify` to find when they added the `IN_MODIFY` event to the package and it seems they have added it long back. I guess we can write the bats test to catch this issue.

@saschagrunert yes, I will take care of this.

**Pull request merged**